src/tagged-type.ts

```diff
@@ -253,9 +284,109 @@ const encode = (input: QueryValue): QueryValue => {
         return encodeMap["set"](input);
       } else if (input instanceof EmbeddedSet) {
         return encodeMap["set"](input);
+      } else if (input instanceof Query) {
+        throw new TypeError(
+          "Cannot encode instance of type 'Query'. Try using TaggedTypeFormat.encodeInterpolation instead."
+        );
       } else {
         return encodeMap["object"](input);
       }
   }
   // anything here would be unreachable code
 };
+
+const encodeInterpolation = (input: QueryValue): QueryInterpolation => {
+  if (input === undefined) {
+    throw new TypeError("Passing undefined as a QueryValue is not supported");
+  }
+  switch (typeof input) {
+    case "bigint":
+      return encodeValueInterpolation(encodeMap["bigint"](input));
+    case "string":
+      return encodeValueInterpolation(encodeMap["string"](input));
+    case "number":
+      return encodeValueInterpolation(encodeMap["number"](input));
+    case "boolean":
+      return encodeValueInterpolation(input);
```

Comment:
can you do a `default: return encode(input)` or something?

Reply:
There is a small, finite list of cases: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/typeof#description

The only cases left were "symbol" and "function", which I just added for completeness. Thanks for catching!